### DISCUSSION TITLE
How do I update the aws-auth ConfigMap - a cutom iam role with a custom group

### DISCUSSION THREAD

**Discussion by sewmiuraj** [gruntwork-io/knowledge-base#688] [How do I]
Hello,

I have created our own custom iam role, that I wish to bind to read-only eks permissions. For this, I have created a custom clusterrolebinding using the existing clusterrole named "view" in the eks cluster. 

We have in the past updated iam roles/users using iam_user_to_rbac_group_mapping = {} and iam_role_to_rbac_group_mapping = {} with groups like ["system:masters"].

But since what I have is a custom group - ["system:devs"], adding this to the eks-cluster's terragrunt.hcl doesn't feel right - since the clusterrolebinding wouldn't be there during initial creation. I'd have preferred if we could have added an if-else clause in the terragrunt.hcl file to only add the group mapping if the cluster/cluster role binding already exists. But I was recommended not to do so in: https://gruntwork-community.slack.com/archives/CHH9Y3Z62/p1679499345840139 

Therefore, I wrote a custom module to separately add the iam_role_to_rbac_group_mapping by directly invoking 
`source_base_url = "git::git@github.com:gruntwork-io/terraform-aws-eks.git//modules/eks-k8s-role-mapping"`

However, in this case it's shows that I am trying to create a new configmap called aws-auth - which is not what I need, but to only update the existing config map. I also don't want my configuration to be removed next time I update the terragrunt.hcl file for eks-cluster.

Is there any way I can achieve this requirement? 

---

<ins datetime="2023-03-24T11:37:40Z">
  <p><a href="https://support.gruntwork.io/hc/requests/110022">Tracked in ticket #110022</a></p>
</ins>

**Comment by pras111gw** [comment]
HI @sewmiuraj have you looked at our `eks-cluster-with-iam-role-mappings` example?
https://github.com/gruntwork-io/terraform-aws-eks/tree/master/examples/eks-cluster-with-iam-role-mappings.

If that doesn't fit your use case, please have a look at terragrunt conditionals that Josh referenced here: https://terragrunt.gruntwork.io/docs/reference/config-blocks-and-attributes/

This page goes over the details of challenges when using conditionals: 
https://blog.gruntwork.io/terraform-tips-tricks-loops-if-statements-and-gotchas-f739bbae55f9#0223
